src_c/camera_windows.c

```diff
@@ -0,0 +1,859 @@
+
+                bgr32_to_rgb(data, surf->pixels, size, surf->format);
+                break;
+            case YUV_OUT:
+                rgb_to_yuv(data, surf->pixels, size,
+                           V4L2_PIX_FMT_XBGR32, surf->format);
+                break;
+            case HSV_OUT:
+                rgb_to_hsv(data, surf->pixels, size,
+                           V4L2_PIX_FMT_XBGR32, surf->format);
+                break;
+        }
+    }
+
+    if (self->pixelformat == MFVideoFormat_YUY2.Data1) {
+        switch (self->color_out) {
+            case YUV_OUT:
+                yuyv_to_yuv(data, surf->pixels, size, surf->format);
+                break;
+            case RGB_OUT:
+                yuyv_to_rgb(data, surf->pixels, size, surf->format);
+                break;
+            case HSV_OUT:
+                yuyv_to_rgb(data, surf->pixels, size, surf->format);
+                rgb_to_hsv(surf->pixels, surf->pixels, size,
+                           V4L2_PIX_FMT_YUYV, surf->format);
+                break;
+        }
+    }
+
+    SDL_UnlockSurface(surf);
+
+    return 1;
+}
+
+int
+windows_read_frame(pgCameraObject *self, SDL_Surface *surf)
+{
+    HRESULT hr;
+
+    if (!self->open) {
+        PyErr_SetString(pgExc_SDLError,
+                        "Camera needs to be started to read data");
+        return 0;
+    }
+
+    if (!_check_integrity(self)) {
+        return 0;
+    };
+
+    if (self->buf) {
+        BYTE *buf_data;
+        DWORD buf_max_length;
+        DWORD buf_length;
+        hr = self->buf->lpVtbl->Lock(self->buf, &buf_data, &buf_max_length,
+                                     &buf_length);
+        CHECKHR(hr);
+
+        if (!windows_process_image(self, buf_data, buf_length, surf)) {
+            return 0;
+        }
+
+        hr = self->buf->lpVtbl->Unlock(self->buf);
+        CHECKHR(hr);
+
+        self->buffer_ready = 0;
+    }
+
+    return 1;
+}
+
+int
+windows_frame_ready(pgCameraObject *self, int *result)
+{
+    *result = self->buffer_ready;
+
+    if (!self->open) {
+        PyErr_SetString(pgExc_SDLError,
+                        "Camera needs to be started to read data");
+        return 0;
+    }
+
+    if (!_check_integrity(self)) {
+        return 0;
+    };
+
+    return 1;
+}
+
+PyObject*
+windows_read_raw(pgCameraObject *self)
+{
+    PyObject* data = NULL;
+    HRESULT hr;
+
+    if (!self->open) {
+        PyErr_SetString(pgExc_SDLError,
+                        "Camera needs to be started to read data");
+        return 0;
+    }
+    
+    if (!_check_integrity(self)) {
+        return 0;
+    };
+
+    if (self->raw_buf) {
+        BYTE *buf_data;
+        DWORD buf_max_length;
+        DWORD buf_length;
+        hr = self->raw_buf->lpVtbl->Lock(self->raw_buf, &buf_data,
+                                         &buf_max_length, &buf_length);
+        CHECKHR(hr);
+
+        data = Bytes_FromStringAndSize(buf_data, buf_length);
```

Comment:
Should this `data` result be checked for error?
https://docs.python.org/3/c-api/bytes.html#c.PyBytes_FromStringAndSize